jpf-symbc/src/main/gov/nasa/jpf/symbc/SymbolicListener.java

```diff
@@ -206,17 +206,37 @@ public void propertyViolated(Search search) {
         // }
     }
 
+    /**
+     * It parses classname and filename to fill the value of assumption.scope at violation witness
+     * Both classname and filename are needed to construct the edge of the witness
+     * @param assumptionScope is a value of assumption.scope attribute of violation witness
+     * @param fileName is a value of originfile attribute of violation witness
+     */
+    public void parseAssumptionScope(ThreadInfo ti){
+        ApplicationContext app = ti.getApplicationContext();
+        String className = app.getMainClassName();
+        String[] parts = className.split("\\.");
+        fileName = parts[parts.length - 1];
+        assumptionScope = String.join(".", parts);
+    }
+
     boolean interceptSymbolic = false;
     boolean fileNameParsed = false;
     String fileName = "";
     String assumptionScope = "";
```

Comment:
why do you have these variables in the middle of the class, should be up when we are defining variables to be used within the class. Having them in the middle is confusing, and considered a bad practice.

Reply:
@sohah , thank you for your advice. I moved these variables to the top. Thank you!